docs/QSG/qemu-vms.md

```diff
@@ -0,0 +1,285 @@
+        --dhcp-leasefile=/var/run/qemu-dnsmasq-$BRIDGE.leases \
+        --dhcp-no-override \
+        ${TFTPROOT:+"--enable-tftp"} \
+        ${TFTPROOT:+"--tftp-root=$TFTPROOT"} \
+        ${BOOTP:+"--dhcp-boot=$BOOTP"}
+    }
+    
+    setup_bridge_nat() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        enable_ip_forward
+        add_filter_rules "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_vlan() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_nat "$BRIDGE"
+    
+    if test "$1" ; then
+        do_ifconfig "$1" 0.0.0.0 up
+        do_brctl addif "$BRIDGE" "$1"
+    fi
+```
+Then add a sudo rule for qemu-system-x86<sub>64</sub> in *etc/sudoers.d*.
+```
+    <username> blackbox=(root) NOPASSWD: /usr/bin/qemu-system-x86_64
+```
+3.  Install guest operating systems.
+
+Run these two commands. Select "boot from cdrom". Follow the installation guide to install the OSes.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on
+```
+4.  Install DAOS software.
+
+After OS installation is finished. Remove the `-cdrom` and `-boot` options from the QEMU commands to boot VMs normally. Next is to install DAOS software.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 
+```
+I follow these [steps](https://docs.daos.io/latest/QSG/setup_rhel/) to install both the DAOS server, DAOS admin, and DAOS client. I install daos-server and daos-admin on the first VM, and install daos-client on the second VM. If you have issues with Ceph Pacific Repo you can edit `/etc/yum.repos.d/CentOS-Ceph-Pacific.repo` by commenting out mirrorlist and changing baseurl as below.
```

Comment:
[root@daosserver01 ~]# lspci | grep -i nvme
**[root@daosserver01 ~]# #this doesn't work all the time as** 
[root@daosserver01 ~]# lspci 
00:00.0 Host bridge: Intel Corporation 82G33/G31/P35/P31 Express DRAM Controller
00:01.0 VGA compatible controller: Device 1234:1111 (rev 02)
00:02.0 Ethernet controller: Red Hat, Inc. Virtio network device
00:03.0 Non-Volatile memory controller: Red Hat, Inc. QEMU NVM Express Controller (rev 02)
00:04.0 SCSI storage controller: Red Hat, Inc. Virtio block device
00:1f.0 ISA bridge: Intel Corporation 82801IB (ICH9) LPC Interface Controller (rev 02)
00:1f.2 SATA controller: Intel Corporation 82801IR/IO/IH (ICH9R/DO/DH) 6 port SATA Controller [AHCI mode] (rev 02)
00:1f.3 SMBus: Intel Corporation 82801I (ICH9 Family) SMBus Controller (rev 02)
**[root@daosserver01 ~]# # As you can see , it is listed as "00:03.0 Non-Volatile memory controller: Red Hat, Inc. QEMU NVM Express Controller (rev 02)"**
[root@daosserver01 ~]# lsblk
NAME        MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sr0          11:0    1 1024M  0 rom  
vda         252:0    0   40G  0 disk 
├─vda1      252:1    0    1G  0 part /boot
└─vda2      252:2    0   39G  0 part 
  ├─rl-root 253:0    0   35G  0 lvm  /
  └─rl-swap 253:1    0    4G  0 lvm  [SWAP]
nvme0n1     259:0    0   16G  0 disk 
[root@daosserver01 ~]#

Reply:
Add a description for the NVMe controller.